packages/providers/src.ts/hedera-provider.ts

```diff
@@ -0,0 +1,130 @@
+import { BaseProvider } from "./base-provider";
+import { AccountBalanceQuery, AccountId, Client, NetworkName, } from '@hashgraph/sdk';
+import { BigNumber } from "@ethersproject/bignumber";
+import { BlockTag, TransactionResponse } from "@ethersproject/abstract-provider";
+import { getAccountFromAddress } from "ethers/lib/utils";
+import axios from 'axios';
+import { Logger } from '@ethersproject/logger';
+import { version } from "./_version";
+
+const logger = new Logger(version);
+
+function sleep(timeout: number) {
+    return new Promise(res => {
+        setTimeout(res, timeout);
+    })
+}
+
+// resolves network string to a hedera network name
+function resolveNetwork(net: string) {
+    switch (net) {
+        case 'mainnet':
+            return NetworkName.Mainnet;
+        case 'previewnet':
+            return NetworkName.Previewnet;
+        case 'testnet':
+            return NetworkName.Testnet;
+        default:
+            logger.throwArgumentError("Invalid network name", "network", net);
+            return null;
+    }
+}
+
+// resolves the mirror node url from the given provider network.
+function resolveMirrorNetworkUrl(net: string): string {
+    switch (net) {
+        case 'mainnet':
+            return 'https://mainnet.mirrornode.hedera.com';
+        case 'previewnet':
+            return 'https://previewnet.mirrornode.hedera.com';
+        case 'testnet':
+            return 'https://testnet.mirrornode.hedera.com';
+        default:
+            logger.throwArgumentError("Invalid network name", "network", net);
+            return null;
+    }
+}
+
+// contains predefined, sdk acceptable hedera network strings
+export enum HederaNetworks {
+    TESTNET = "testnet",
+    PREVIEWNET = "previewnet",
+    MAINNET = "mainnet"
+}
+
+/**
+ * The hedera provider uses the hashgraph module to establish a connection to the Hedera network.
+ * As every provider, this one also gives us read-only access.
+ *
+ * Constructable with a string, which automatically resolves to a hedera network via the hashgraph SDK.
+ */
+export class DefaultHederaProvider extends BaseProvider {
+    private readonly hederaClient: Client;
+    private readonly hederaNetwork: string;
```

Comment:
The `baseProvider` has this info as parameter. I don't think that we need it here